### ISSUE TITLE
Total failure

### ISSUE THREAD

**Issue by jmcd040** [writemonkey/wm3#308]
I have numerous files both in the database and on disk. I added a new scratch document with the plus sign, pasted some text into it and later deleted the text. A little while later when I came back to WM3, all my work was gone. My project names were there, but contained no documents. The ALL file selection showed no documents, just several empty SCRATCH documents. I checked the database backup and it matches the current database. So no help there. Somehow the database backup had been cleaned out too.

I was finally able to recover database backup files that were about 3 hours old from my Windows File History and restore the active database files from there. But that puts me hours behind.
I'm on a tight deadline and this was disturbing.

**Comment by writemonkey** [comment]
Fuf, that sucks. I don't know how is that even possible. WM won't allow multiple instances of the app so multiple nw.exes is also very strange.
i.

**Comment by jmcd040** [comment]
I later noticed that when WM3 runs, it creates multiple tasks, so maybe it wasn't actually multiple instances. However the program wouldn't close so I had to kill all the tasks one by one in task manager. I wish I had paid attention more to exactly what the steps were I took in all this.

**Comment by jmcd040** [comment]
It happened again. I had WM3 save backups several times yesterday. All was working normally. I closed the program last night. This morning there were no files - project definitions and repository were ok but no files. By looking at my Windows File History which backs up every 30 minutes, I found that the writemonkey3_sheets file had vanished sometime yesterday afternoon. Fortunately I was able to find an older backup and WM3 updated my working file from my hard drive.

**Comment by jmcd040** [comment]
Happened again. Was working on several docs. Condensed and backed up the database, but noticed that the menu text indicating the from the last backup wouldn't change. Closed the program. Opened and all the documents were gone. The sheets file was empty. Restored from a backup and decided to dump everything from the database except the file I'm working on and some docs that are not linked to external files that I don't modify often, so I don't care if they are restored from old backups. I don't find this a comforting way to work.

**Comment by writemonkey** [comment]
Hi Jim, unfortunately you are the only one reporting this. I took some time and tried to corrupt the database by compacting  and backuping the db multiple times in fast succession. No corruption at my end.

- How big are your db files? writemonkey3_sheets and writemonkey3_sheets_history?
- Have you changed the default path for db files?
- What is your setting for auto save (`auto_save_interval`)?
- Is `keep_document_undo_data` set to false?
- How big was the last opened file when error occurred? Did you work on any very large documents when  this happened?

i.

**Comment by jmcd040** [comment]
Size of files: 
sheets 642kb, sheets-history 212 Mb
I set the backup directory, but not the db directory location
auto_save_interval is 5
keep_document_undo_data is the default, false
My documents aren't large, say 3 to 120Kb
backup_database_automatically is set to 1
I have had about 25 documents.

**Comment by DaceKonn** [comment]
> Fuf, that sucks. I don't know how is that even possible. WM won't allow multiple instances of the app so multiple nw.exes is also very strange.
> i.

Actually it's default:

![image](https://user-images.githubusercontent.com/22869861/65506707-7b030180-decc-11e9-98fe-662ed89698f0.png)

I'm guessing that's a default nodejs behavior or something - one process for each thread or something? I'm not a nodejs specialist. But that's a default way how the app works.

**Comment by writemonkey** [comment]
Yes, my bad, on Windows, multiple nw.exes in tasks list are ok. Each process is represented with one nw.exe. There should be 4 of them.
i.
